stock_picking_product_barcode_report/report/report_label_barcode_template.xml

```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <template id="report_label_barcode">
+        <t t-call="web.basic_layout">
+            <div class="page text-justify">
+                <t t-foreach="doc.label_qty" t-as="i">
+                    <div class="text-center" t-if="doc.product_id.barcode">
+                        <t
+                            t-if="doc.env.user.company_id.barcode_default_format == 'gs1_128'"
+                        >
+                            <t t-if="doc.move_line_id.lot_id">
+                                <img
+                                    t-att-src="'/report/barcode/?type=%s&amp;value=%s&amp;width=%s&amp;height=%s' % ('gs1_128', '02' + doc.product_id.barcode.zfill(14) + '10' + doc.move_line_id.lot_id.name, 600, 100)"
+                                    style="width:100%;height:70px;margin: 0px 0px -25px 0px"
+                                    alt="Barcode"
+                                />
+                                <span
+                                    t-esc="'(02)' + doc.product_id.barcode.zfill(14) + '(10)' + doc.move_line_id.lot_id.name"
```

Comment:
create a variable to store this and you can save a lot code lines.
Also you can store type in a variable...